### PR TITLE
Allow setting failSwapOn flag in the shoot.yaml file

### PR DESCRIPTION
**Pull request by saggir**
How to categorize this PR?

/area performance
/area documentation
/kind enhancement
/priority normal

What this PR does / why we need it:
This PR adds support for failSwapOn flag in the shoot yaml file, as part of the kubelet section

Today this parameter is set to true by default and there is no way to change it, adding the support in the shoot yaml will alow changing it to "false".
The default value is still "true"

From Kubernetes documentation: 
--fail-swap-on:
Makes the Kubelet fail to start if swap is enabled on the node. (default true) (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)

Release note:
`Shoot resource now supports changing the failSwapOn flag under the kubelet section, default=true`

### CODE REVIEW

File: pkg/apis/core/v1alpha1/types_shoot.go [hunk 1]
```diff
@@ -626,6 +626,9 @@ type KubeletConfig struct {
 	// +optional
 	// Default: 1m
 	ImagePullProgressDeadline *metav1.Duration `json:"imagePullProgressDeadline,omitempty" protobuf:"bytes,12,opt,name=imagePullProgressDeadline"`
+	// FailSwapOn makes the Kubelet fail to start if swap is enabled on the node. (default true).
+	// +optional
+	FailSwapOn *bool `json:"failSwapOn,omitempty" protobuf:"varint,13,opt,name=failSwapOn"`
```
Comment (rfranzke) [inline pkg/apis/core/v1alpha1/types_shoot.go:631]: Should we default it to `true` in the `defaults.go`?

File: pkg/apis/core/v1beta1/types_shoot.go [hunk 2]
```diff
@@ -623,6 +623,9 @@ type KubeletConfig struct {
 	// +optional
 	// Default: 1m
 	ImagePullProgressDeadline *metav1.Duration `json:"imagePullProgressDeadline,omitempty" protobuf:"bytes,12,opt,name=imagePullProgressDeadline"`
+	// FailSwapOn makes the Kubelet fail to start if swap is enabled on the node. (default true).
+	// +optional
+	FailSwapOn *bool `json:"failSwapOn,omitempty" protobuf:"varint,13,opt,name=failSwapOn"`
```
Comment (rfranzke) [inline pkg/apis/core/v1beta1/types_shoot.go:628]: Should we default it to `true` in the `defaults.go`?

**Review by rfranzke — CHANGES_REQUESTED** [review]
Thanks for the PR!

1. Can you please fix the file permissions?
1. Can you please squash the commits?

**Comment by vpnachev** [comment]
Fixes #2120 ?

File: pkg/apis/core/v1alpha1/types_shoot.go (hunk 1 above)
Comment (saggir) [reply pkg/apis/core/v1alpha1/types_shoot.go:631]: I dont think it is needed, it behaves the same as CPUCFSQuota, which dont have an entry in the defaults.go file

File: pkg/apis/core/v1beta1/types_shoot.go (hunk 2 above)
Comment (saggir) [reply pkg/apis/core/v1beta1/types_shoot.go:628]: I dont think it is needed, it behaves the same as CPUCFSQuota, which dont have an entry in the defaults.go file

**Comment by saggir** [comment]
replaced with a cleaner PR - https://github.com/gardener/gardener/pull/2411

**Pull request closed without merging**